source/dev/idea_list.md

```diff
@@ -0,0 +1,148 @@
+
+
+- Write Apache Airflow workflows that periodically 
+    - Query external data sources for availability
+    - Download external data sources
+    - Validate and clean the downloaded resources
+    - Archive downloaded resources to a blob store
+- Workflows are instrumented with telemetry
+- Telemetry are aggregated in InfluxDB
+- Metrics are displayed in a Grafana dashboard
+- If a data source is down, we throw an alarm
+The workflows should be well documented with a focus on code quality, and using best-of-breed current technology stacks (Apache Airflow, Kubernetes, Helm, Docker, Python3, Grafana, etc.)
+
+**Requirements:**
+Docker, Python, Grafana, PAAS, Apache Airflow
+
+**Level:** 
+Medium to Hard
+
+**Mentor:**
+
+### Project 4. - Test system for end-to-end testing of the simulation framework 
+Use [existing test power systems](http://labs.ece.uw.edu/pstca/) that will be used in our simulation framework. 
+Generate input data to feed to the simulation engine using data science techniques.
+Write integration tests and produce documentation to showcase our open source software.
+
+- Collect data (python scripting)
+- Generate time series for demand, hydro, solar and wind matching the test grid (python data analysis)
+- Integrate the test model in our [simulation framework](https://github.com/Breakthrough-Energy/PowerSimData) (python)
+- Write end-to-end tests (in docker container)
+- Write GitHub actions to test changes of the framework when pushed on GitHub
+- Write documentation for the community (sphinx)
+
+**Requirements:**
+Object Oriented Programming in Python, Data analysis/plotting in Python (pandas, numpy, matplotlib, etc.) and docker
+
+**Level:** 
+Easy
+
+**Mentor:**
+[Ben](https://github.com/rouille)
+
+### Project 5. - Time-Series Modeling of Missing Energy Industry Data
+A significant barrier to realistic modeling of the power grid is missing data from public data sources.
+This project is to use statistical and ML models to fill in missing energy demand and renewable energy profile information in a way which mimics the real data as close as possible.
+In our current datasets of regional energy demand, we have used averaging of adjacent days to fill in our missing data, as described [here](https://github.com/Breakthrough-Energy/PreREISE#d-demand-data)
+
+We propose the following steps for this project:
+
+
+- Understand the data set and the [current methodology](https://github.com/Breakthrough-Energy/PreREISE#d-demand-data)
+- Literature review to understand prediction models for time series data
+- Collect data to build training/validation/testing sets to rank algorithms using various metrics
+- Setup platform to crossvalidate with the data in order to evaluate algorithm performance
+- Integrate most suitable algorithm(s) in our code base and generate new time series for our power system model
```

Comment:
How will this interpolation fit into the data ingestion pipeline?

Reply:
The ML algorithms will be used to improve the imputation of missing data. Those, along with the existing one in PreREISE, could be automatically called in Project 3

Reply:
This more that just interpolation, the idea is to leverage our existing data better to create a model to fill in the gaps. Interpolation is a more localized operation based on adjacent data points. There are some pretty huge gaps in the datasets which can’t be fixed with simple interpolation.

Reply:
@dmuldrew - is there a word that describes "fixing huge gaps in datasets" that we can use here?